### PR TITLE
[GLUTEN-1336][VL] add spark3.3 UT under connector and expression

### PR DESCRIPTION
**Pull request by yma11**
## What changes were proposed in this pull request?

Port vanilla spark 3.3 UT to Gluten, includes the ones under connector and expression

## How was this patch tested?

unit tests

### CODE REVIEW

**Comment by zhouyuan** [comment]
Run Gluten Clickhouse CI

File: gluten-ut/spark33/src/test/scala/io/glutenproject/utils/velox/VeloxTestSettings.scala [hunk 1]
```diff
@@ -18,12 +18,149 @@
...
+    .exclude("random")
+    .exclude("SPARK-9127 codegen with long seed")
+  enableSuite[GlutenRegexpExpressionsSuite]
+  enableSuite[GlutenSortOrderExpressionsSuite]
+  enableSuite[GlutenStringExpressionsSuite]
+    .exclude("Substring")
+    .exclude("string for ascii")
+    .exclude("replace")
+    // ascii for Latin-1 function return from Velox is not correct
+    .exclude(("SPARK-40213: ascii for Latin-1 Supplement characters"))
```
Comment (PHILO-HE) [inline gluten-ut/spark33/src/test/scala/io/glutenproject/utils/velox/VeloxTestSettings.scala:165]: There is a fix PR for this UT. See https://github.com/oap-project/gluten/pull/1715. Please remove the exclusion after that PR is merged.

Comment (PHILO-HE) [reply gluten-ut/spark33/src/test/scala/io/glutenproject/utils/velox/VeloxTestSettings.scala]: FYI. The fix for ascii func. has been merged.

**Review by zhouyuan — APPROVED** [review]
👍

**Comment by PHILO-HE** [comment]
Please rebase your code to fix the CI failure.

**Review by PHILO-HE — APPROVED** [review]
LGTM

**Comment by PHILO-HE** [comment]
Currently, spark 3.3 tests are executed by CI job `velox-backend-ubuntu2004-test-slow`. I think we can move them into a separate CI job in the future. cc @zhouyuan

**Comment by zhouyuan** [comment]
> Currently, spark 3.3 tests are executed by CI job `velox-backend-ubuntu2004-test-slow`. I think we can move them into a separate CI job in the future. cc @zhouyuan

indeed, let's follow up in another patch. 

-yuan

**Pull request merged**